pkg/lib/infra/sms/config_resolver.go

```diff
@@ -0,0 +1,118 @@
+package sms
```

Comment:
I suggest we have client_resolver.go only. You unify config_resolver and provider_resolver and make them client_resolver.